# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1765576082 -3600
# Parent  a370617ed5f171c0bb5e5818696d11481643e6f6
Backout bug 1966100 for breaking menu popups on Windows in 140.

diff --git a/toolkit/themes/shared/menu.css b/toolkit/themes/shared/menu.css
--- a/toolkit/themes/shared/menu.css
+++ b/toolkit/themes/shared/menu.css
@@ -359,27 +359,40 @@ menuitem[highlightable] > .menu-text {
 }
 
 /* On windows we use a checkmark */
 @media (-moz-platform: windows) {
-  /* We need to do something to override the default style for selected
-     checkboxes and radio buttons because the native drawing we use on other
-     Windows versions looks pretty bad with the Win10 styles, so for now we'll
-     insert a generic checkmark icon for both types. */
-  menuitem[checked="true"] > .menu-icon {
-    -moz-context-properties: fill, fill-opacity;
-    fill: currentColor;
-    fill-opacity: var(--menu-icon-opacity);
-    list-style-image: url("chrome://global/skin/icons/menu-check.svg");
+  menuitem[checked="true"] {
+    padding-inline-start: 12px;
+
+    /* We need to do something to override the default style for selected
+       checkboxes and radio buttons because the native drawing we use on other
+       Windows versions looks pretty bad with the Win10 styles, so for now we'll
+       insert a generic checkmark icon for both types. */
+    > .menu-icon {
+      display: revert;
+      -moz-context-properties: fill, fill-opacity;
+      fill: currentColor;
+      fill-opacity: var(--menu-icon-opacity);
+      list-style-image: url("chrome://global/skin/icons/menu-check.svg");
+      margin-inline-end: 8px;
+    }
   }
 }
 
 menupopup[needsgutter] {
-  @media (-moz-platform: linux) or (-moz-platform: windows) {
+  @media (-moz-platform: linux) {
     > :is(menu, menuitem) > .menu-icon {
       display: revert;
     }
   }
 
+  @media (-moz-platform: windows) {
+    > menu:not([icon], .menu-iconic),
+    > menuitem:not([checked="true"], [icon], .menuitem-iconic) {
+      padding-inline-start: 32px;
+    }
+  }
+
   @media (-moz-platform: macos) {
     /* although only menuitems may be checked, apply this to
        menus and menucaptions as well to maintain alignment */
     > menu,
